src/apps/replication/lib/mutation_log.cpp

```diff
@@ -564,7 +565,8 @@ void mutation_log::internal_write_callback(
 
     if (err == ERR_OK || err == ERR_HANDLE_EOF)
     {
-        dassert(g_end_offset == end_offset,
+        // the log may still be written when used for learning
```

Comment:
I think sync checkpoint stalling itself from committing the local app makes world difficult, maybe can we not consider(support) it?